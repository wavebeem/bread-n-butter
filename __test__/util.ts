import * as bnb from "../src/bread-n-butter";

export function snapTest<A>(parser: bnb.Parser<A>, input: string): void {
  expect(parser.parse(input)).toMatchSnapshot(JSON.stringify(input));
}
