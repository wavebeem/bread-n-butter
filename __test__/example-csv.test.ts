import CSV from "../examples/csv";
import { snapTest } from "./util";

test("csv simple", () => {
  snapTest(
    CSV,
    `\
apple,1.23,1312
banana,0.99,67
cherry,0.54,987
dragonfruit,2.87,4
elderberry,9.99,22
`
  );
});

test("csv complex", () => {
  snapTest(
    CSV,
    `\
a,,c,"a ""complex"" field, i think"
d,eeeeee,FFFF,cool
nice,nice,nice3,nice4
`
  );
});

test("csv LF", () => {
  snapTest(
    CSV,
    `\
a,b,c\n\
a,b,c\n\
a,b,c\
`
  );
});

test("csv CRLF", () => {
  snapTest(
    CSV,
    `\
a,b,c\r\n\
a,b,c\r\n\
a,b,c\
`
  );
});

test("csv trailing newline", () => {
  snapTest(
    CSV,
    `\
a,b,c\r\n\
a,b,c\r\n\
a,b,c\r\n\
`
  );
  snapTest(
    CSV,
    `\
a,b,c\n\
a,b,c\n\
a,b,c\n\
`
  );
});
