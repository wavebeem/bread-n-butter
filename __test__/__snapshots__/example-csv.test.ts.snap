// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csv CRLF: "a,b,c\\r\\na,b,c\\r\\na,b,c" 1`] = `
Object {
  "type": "ParseOK",
  "value": Array [
    Array [
      "a",
      "b",
      "c",
    ],
    Array [
      "a",
      "b",
      "c",
    ],
    Array [
      "a",
      "b",
      "c",
    ],
  ],
}
`;

exports[`csv LF: "a,b,c\\na,b,c\\na,b,c" 1`] = `
Object {
  "type": "ParseOK",
  "value": Array [
    Array [
      "a",
      "b",
      "c",
    ],
    Array [
      "a",
      "b",
      "c",
    ],
    Array [
      "a",
      "b",
      "c",
    ],
  ],
}
`;

exports[`csv complex: "a,,c,\\"a \\"\\"complex\\"\\" field, i think\\"\\nd,eeeeee,FFFF,cool\\nnice,nice,nice3,nice4\\n" 1`] = `
Object {
  "type": "ParseOK",
  "value": Array [
    Array [
      "a",
      "",
      "c",
      "a \\"complex\\" field, i think",
    ],
    Array [
      "d",
      "eeeeee",
      "FFFF",
      "cool",
    ],
    Array [
      "nice",
      "nice",
      "nice3",
      "nice4",
    ],
  ],
}
`;

exports[`csv simple: "apple,1.23,1312\\nbanana,0.99,67\\ncherry,0.54,987\\ndragonfruit,2.87,4\\nelderberry,9.99,22\\n" 1`] = `
Object {
  "type": "ParseOK",
  "value": Array [
    Array [
      "apple",
      "1.23",
      "1312",
    ],
    Array [
      "banana",
      "0.99",
      "67",
    ],
    Array [
      "cherry",
      "0.54",
      "987",
    ],
    Array [
      "dragonfruit",
      "2.87",
      "4",
    ],
    Array [
      "elderberry",
      "9.99",
      "22",
    ],
  ],
}
`;

exports[`csv trailing newline: "a,b,c\\na,b,c\\na,b,c\\n" 1`] = `
Object {
  "type": "ParseOK",
  "value": Array [
    Array [
      "a",
      "b",
      "c",
    ],
    Array [
      "a",
      "b",
      "c",
    ],
    Array [
      "a",
      "b",
      "c",
    ],
  ],
}
`;

exports[`csv trailing newline: "a,b,c\\r\\na,b,c\\r\\na,b,c\\r\\n" 1`] = `
Object {
  "type": "ParseOK",
  "value": Array [
    Array [
      "a",
      "b",
      "c",
    ],
    Array [
      "a",
      "b",
      "c",
    ],
    Array [
      "a",
      "b",
      "c",
    ],
  ],
}
`;
